src/bin/pgcopydb/env_utils.c

```diff
@@ -116,6 +116,44 @@ get_env_copy_with_fallback(const char *name, char *result, int maxLength,
 	return true;
 }
 
+/*
+ * get_env_dup copies the environment variable with "name" into
+ * the result buffer using strdup. It returns false when it fails. If the environment
+ * variable is not set the fallback string will be written in the buffer.
+ * Except when fallback is NULL, in that case an error is returned.
+ */
+bool
+get_env_dup(const char *name, char **result, const char *fallback)
+{
+	if (name == NULL || strlen(name) == 0)
+	{
+		log_error("Failed to get environment setting. "
+				  "NULL or empty variable name is provided");
+		return false;
+	}
+
+	/*
+	 * Explanation of IGNORE-BANNED
+	 * getenv is safe here because we never provide null argument,
+	 * and copy out the result immediately.
+	 */
+	const char *envvalue = getenv(name); /* IGNORE-BANNED */
+	if (envvalue == NULL)
+	{
+		envvalue = fallback;
+		if (envvalue == NULL)
+		{
+			log_error("Failed to get value for environment variable '%s', "
+					  "which is unset", name);
+			return false;
+		}
+	}
+	*result = strdup(envvalue);
```

Comment:
This requires testing the result value of the `strdup` call, which might return NULL when memory allocation failed. See https://man7.org/linux/man-pages/man3/strdup.3.html.